backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,148 @@
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.DurationUnit
+
+@Composable
+fun BpkFloatingNotification(
+  modifier: Modifier = Modifier,
+  text: String,
+  show: Boolean,
+  icon: BpkIcon? = null,
+  animation: Animation = Animation(),
+  cta: CTA? = null
+) {
+  val (internalShow, setInternalShow) = remember { mutableStateOf(false) }
+  if (show) {
+    LaunchedEffect(key1 = Unit) {
+      setInternalShow(true)
+      delay(animation.showDuration)
+      setInternalShow(false)
+      animation.onFinished?.invoke()
+    }
+  }
+  val animationDuration = if (animation.enabled) animation.transitionDuration.toInt(DurationUnit.MILLISECONDS) else 0
+  val fadeAnimationSpec: FiniteAnimationSpec<Float> = tween(durationMillis = animationDuration, easing = LinearEasing)
+  val slideAnimationSpec: FiniteAnimationSpec<IntOffset> = tween(durationMillis = animationDuration, easing = LinearEasing)
+  AnimatedVisibility(
+    visible = internalShow,
+    enter = slideInVertically(slideAnimationSpec, initialOffsetY = { it / 2 }) + fadeIn(animationSpec = fadeAnimationSpec),
+    exit = slideOutVertically(slideAnimationSpec, targetOffsetY = { it / 2 }) + fadeOut(animationSpec = fadeAnimationSpec)
+  ) {
+    Box(
+      modifier = modifier
+        .requiredHeightIn(min = 52.dp, max = 72.dp)
+        .requiredWidthIn(min = 288.dp, max = 400.dp)
+        .padding(start = BpkSpacing.Base, end = BpkSpacing.Base),
+      contentAlignment = Alignment.BottomCenter
+    ) {
+      BpkCard(
+        modifier = modifier.fillMaxWidth(),
+        backgroundColor = BpkTheme.colors.corePrimary
+      ) {
+        Row(
+          modifier = Modifier.wrapContentSize(),
+          horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+          verticalAlignment = Alignment.CenterVertically
+        ) {
+          icon?.let { icon ->
+            BpkIcon(
+              modifier = Modifier,
+              icon = icon,
+              contentDescription = "",
+              size = BpkIconSize.Small,
+              tint = BpkTheme.colors.textOnDark
+            )
+          }
+          BpkText(
+            modifier = Modifier.weight(0.8f),
+            text = text,
+            color = BpkTheme.colors.textOnDark,
+            style = BpkTheme.typography.footnote
+          )
+          cta?.let {
+            Box(
+              modifier = Modifier
+                .requiredSize(BpkSpacing.Xxl)
+                .clip(CircleShape)
+                .weight(0.1f)
+                .clickable { cta.onClick.invoke() },
+              contentAlignment = Alignment.Center
+            ) {
+              BpkText(
+                text = cta.text,
+                color = BpkTheme.colors.textOnDark,
+                style = BpkTheme.typography.label2
+              )
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+data class CTA(
+  val text: String,
+  val onClick: (() -> Unit)
```

Comment:
No need to have extra braces here. Also, we tend to use trailing commas when possible in Backpack